packages/shopify-api/lib/session/session.ts

```diff
@@ -85,55 +87,73 @@ export class Session {
                 },
               ];
             case 'userId':
-              return [
-                key,
-                (associatedUserObj.associated_user.id = Number(value)),
-              ];
+              if (returnUserData) {
```

Comment:
This switch statement got a more complicated with this change. Let me know if you are okay with this, or you would like this refactored into a (possibly) more readable if/else

Reply:
I think that's probably fine - this will eventually be removed so it's not the end of the world.